homeassistant/components/watttime/config_flow.py

```diff
@@ -0,0 +1,237 @@
+"""Config flow for WattTime integration."""
+from __future__ import annotations
+
+import asyncio
+from typing import TYPE_CHECKING, Any
+
+from aiowatttime import Client
+from aiowatttime.errors import (
+    CoordinatesNotFoundError,
+    InvalidCredentialsError,
+    UsernameTakenError,
+)
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import (
+    CONF_EMAIL,
+    CONF_LATITUDE,
+    CONF_LONGITUDE,
+    CONF_PASSWORD,
+    CONF_USERNAME,
+)
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.helpers import aiohttp_client, config_validation as cv
+
+from .const import (
+    AUTH_TYPE_LOGIN,
+    AUTH_TYPE_REGISTER,
+    CONF_BALANCING_AUTHORITY,
+    CONF_BALANCING_AUTHORITY_ABBREV,
+    CONF_BALANCING_AUTHORITY_ID,
+    DOMAIN,
+    LOGGER,
+)
+
+CONF_AUTH_TYPE = "auth_type"
+CONF_ORGANIZATION = "organization"
+
+STEP_LOGIN_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_USERNAME): str,
+        vol.Required(CONF_PASSWORD): str,
+    }
+)
+
+STEP_REGISTER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_USERNAME): str,
+        vol.Required(CONF_PASSWORD): str,
+        vol.Required(CONF_EMAIL): str,
+        vol.Required(CONF_ORGANIZATION): str,
+    }
+)
+
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_AUTH_TYPE): vol.In([AUTH_TYPE_LOGIN, AUTH_TYPE_REGISTER]),
```

Comment:
Let's not support registration. Let users do that on the website.

Reply:
You can't register via the website; you can only register via an API request. https://www.watttime.org/api-documentation/#authentication

Reply:
We shouldn't include account management in Home Assistant. We want to integrate with services, not provide a full interface for it.